examples/all-clusters-app/all-clusters-common/src/operational-state-delegates.cpp

```diff
@@ -0,0 +1,101 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app/clusters/operational-state-server/operational-state-server.h>
+#include <app/util/config.h>
+#include <operational-state-delegate-impl.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace OperationalState {
+
+/**
+ * class to present Enquriy Table of Delegates
+ */
+struct DelegatesEnquiryTable
+{
+    /**
+     * Endpoint Id
+     */
+    chip::EndpointId mEndpointId;
+    /**
+     * Cluster Id
+     */
+    chip::ClusterId mClusterId;
+    /**
+     * point of Array(Items)
+     */
+    Delegate * pItems;
+    /**
+     * ArraySize of Array(Items)
+     */
+    size_t numOfItems;
+};
+
+/**
+ * Enquriy Table of Operational State Delegate
+ * Note: User Define
+ */
+static OperationalStateDelegate opStateDelegate(1, Clusters::OperationalState::Id,
+                                                GenericOperationalState(to_underlying(OperationalStateEnum::kStopped)),
+                                                GenericOperationalError(to_underlying(ErrorStateEnum::kNoError)));
+
+/**
+ * Enquriy Table of Operational State Cluster and alias Cluter Delegate corresponding to endpointId and clusterId
+ * Note: User Define
+ */
+constexpr DelegatesEnquiryTable kDelegatesEnquiryTable[] = {
+    // EndpointId, ClusterId, Delegate
+    { 1, Clusters::OperationalState::Id, &opStateDelegate },
+};
+
+/**
+ * Get the pointer of target delegate for target endpoint and cluster
+ * @param[in] aEndpiontId The endpointId
+ * @param[in] aClusterID  The clusterId
+ * @return the pointer of target delegate
+ */
+Delegate * getGenericDelegateTable(chip::EndpointId aEndpointId, chip::ClusterId aClusterId)
```

Comment:
is this protected or private helper ?
seems to be a very straightforward function among server

Reply:
I imitate  the api `DefaultSceneTableImpl * GetSceneTableImpl(EndpointId endpoint, uint16_t endpointTableSize)` in `src/app/clusters/scenes-server/SceneTableImpl.cpp` , So is there any better implement for delegate? Can you give some suggestion? Thanks.